### ISSUE TITLE
Seeing multiple bootstrap contexts with Spring Boot hierarchy

### ISSUE THREAD

**Issue by bijukunjummen** [spring-cloud/spring-cloud-commons#153]
There is likely a gap in my understanding here. When I create a spring cloud based application with a hierarchy, say the following:

```java
SpringApplicationBuilder appBuilder =
                new SpringApplicationBuilder()
                        .parent(RootContext.class)
                        .child(ChildContext.class);

        ConfigurableApplicationContext applicationContext  = appBuilder.run();
```

I am seeing **two bootstrap** contexts, visualized here using the `/beans` actuator endpoint:

<img width="479" alt="screen shot 2016-12-17 at 1 55 19 pm" src="https://cloud.githubusercontent.com/assets/390605/21290009/93d93d66-c460-11e6-8f57-baf4b6c656a3.png">

My expectation was to see exactly one bootstrap context.

**Comment by spencergibb** [comment]
@dsyer any thoughts

**Comment by dsyer** [comment]
Fixed in 331e790
